src/adapters/sonobi.js

```diff
@@ -13,36 +13,51 @@ var SonobiAdapter = function SonobiAdapter(){
     adloader.loadScript(trinity + JSON.stringify(_keymaker(bids)) + '&cv=' + _operator(), null);
   }
 
-  function _keymaker(bids){       //  Make keys
+  function _keymaker(bids){               //  Make keys
     var keyring = {};
     utils._each(bids, function(o){
-      var sizes = [];
-      utils._each(o.sizes, function(size){
-        sizes.push(size.join('x'));
-      });
-      sizes = sizes.toString();
-      if (!!o.params.ad_unit && o.params.ad_unit.length > 0) {
-        //  Cypher
-        keyring[o.params.ad_unit + '|' + o.params.dom_id] = sizes;
-        cb_map[o.params.ad_unit + '|' + o.params.dom_id] = o.placementCode;
-      } else if (!!o.params.placement_id && o.params.placement_id.length > 0) {
-        //  Morpheus
-        keyring[o.params.dom_id] = o.params.placement_id + (test ? '-test' : '')  + '|' + sizes;
-        cb_map[o.params.dom_id] = o.placementCode;
-      } else {
-        utils.logError('sonobi unable to bid: Improper parameters for ' + o.placementCode);
+      var sizes = utils.parseSizesInput(o.sizes).toString();
+      if (utils.isEmpty(sizes)){
+        utils.logWarn('Sonobi adapter expects sizes for ' + o.placementCode);
+      }
+      switch(true){
+        case (!o.params.ad_unit && !o.params.placement_id):
+          utils.logError('Sonobi unable to bid: Missing parameters for ' + o.placementCode);
+          break;
+        case (!!o.params.ad_unit && !!o.params.placement_id):
+          utils.logError('Sonobi unable to bid: Extra parameters for ' + o.placementCode);
+          break;
+        case (!!o.params.ad_unit && o.params.ad_unit.length === 0):
+          utils.logError('Sonobi unable to bid: Empty ad_unit for ' + o.placementCode);
+          break;
+        case (!!o.params.placement_id && o.params.placement_id.length === 0):
+          utils.logError('Sonobi unable to bid: Empty placement_id for ' + o.placementCode);
+          break;
+        case (!!o.params.placement_id):     //  Morpeus style 
+          keyring[o.params.dom_id] = o.params.placement_id + (test ? '-test' : '')  + '|' + sizes;
+          cb_map[o.params.dom_id] = o.placementCode;
+          break;
+        case (!!o.params.ad_unit && o.params.ad_unit.charAt(0) !== '/'):
+          //  DFP docs do not necessarily require leading slash? - add it in if it's not there.
+          o.params.ad_unit = '/' + o.params.ad_unit;
```

Comment:
Yes, this is intentional. The action should continue with the fall through.

The jslint library does not recognize this, which is incorrect - this is both valid and intentional.